tt_metal/impl/device/device.cpp

```diff
@@ -1342,6 +1345,10 @@ void Device::compile_command_queue_programs() {
     constexpr uint32_t prefetch_d_downstream_cb_sem = 2;
     constexpr uint32_t dispatch_downstream_cb_sem = 1;
 
+    // TODO: this->hw_command_queues_[cq_id]->noc_index is also hardcoded to NOC_0 elsewhere, should have one definition
```

Comment:
does this need to be checked with some assert?

Reply:
Yeah I'll add an assert for this + the downstream noc must be different from the default noc.